tests/Unit/ProtoTests/DeprecatedService/out/src/Client/BaseClient/DeprecatedServiceBaseClient.php

```diff
@@ -142,9 +146,27 @@ public function __construct(array $options = [])
         $this->setClientOptions($clientOptions);
     }
 
+    public function __call($method, $args)
+    {
+        if (substr($method, -5) !== 'Async') {
+            throw new ValidationException("Method name $method does not exist");
```

Comment:
We probably want to imitate PHP's method not foundd error here, if someone calls a random method on our client, they'll get an unexpected error. So we'll want something like

```suggestion
            trigger_error('Call to undefined method '.__CLASS__.'::'.$method.'()', E_USER_ERROR);
```

Reply:
Ok this is way better, thank you. Will do.

Reply:
Done. Used an interpolated string for the `::$method()` part though.

Reply:
perfect!